crates/nargo/src/cli/fs/program.rs

```diff
@@ -0,0 +1,39 @@
+use std::path::{Path, PathBuf};
+
+use acvm::hash_constraint_system;
+use noirc_driver::CompiledProgram;
+
+use crate::errors::CliError;
+
+use super::{create_named_dir, write_to_file};
+
+pub(crate) fn save_program_to_file<P: AsRef<Path>>(
+    compiled_program: &CompiledProgram,
+    circuit_name: &str,
+    circuit_dir: P,
+) -> PathBuf {
+    let mut circuit_path = create_named_dir(circuit_dir.as_ref(), "target");
+    circuit_path.push(circuit_name);
+    circuit_path.set_extension("json");
+
+    write_to_file(&serde_json::to_vec(compiled_program).unwrap(), &circuit_path);
+
+    // Save a checksum of the circuit to compare against during proving and verification
+    let acir_hash = hash_constraint_system(&compiled_program.circuit);
+    circuit_path.set_extension("json.sha256");
```

Comment:
This seems duplicated?

Reply:
Duplicated in what way? The only other time we interact with this is to read it along with the pk/vk

Reply:
Oh hmm its just the extension part; in fetch_pk_and_vk, we also do:

```
acir_hash_path.set_extension("json.checksum");
```

Not a blocker as it was there before